### ISSUE TITLE
[Feature][Dashboard] Use month of merged_date to group for "mean time to merge pull requests"

### ISSUE THREAD

**Issue by asterix314** [apache/incubator-devlake#2674]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

In the "GitHub Basic Metrics" dashboard, the calculation of "mean time to merge pull requests" is anchored on the PR _creation date_,  as per Inspect/Query, or in simplified SQL:

```sql
SELECT TRUNCATE_MONTH(created_date) AS created_month, AVG(merged_date - created_date) AS time_to_merge
FROM pull_requests
GROUP BY created_month
```

The resulting bar chart would most likely have a downward trend towards the right end, such as the following:

<img width="797" alt="image" src="https://user-images.githubusercontent.com/10952409/182653088-1eda2251-7c6b-4191-b7a0-e29518fc6333.png">

It is because the time to merge must by necessity be smaller than the PR's age, so young PR's appear to be merged sooner than old ones on average. A bias on the PR's age is thus introduced.

This feature request asks that the same calculation be anchored on the PR _merge date_ instead:

```sql
SELECT TRUNCATE_MONTH(merged_date) AS merged_month, AVG(merged_date - created_date) AS time_to_merge
FROM pull_requests
GROUP BY merged_month
```

which should get rid of the bias.

Note the same bias also manifests in "mean time to close pull requests", and potentially in other dashboards, too, where an average of some interval is calculated anchoring on the beginning time.

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Startrekzky** [comment]
Good catch @asterix314 . I noticed this problem and got it fixed in v0.12.0. May I know the version number you're using?

**Comment by asterix314** [comment]
Hi @Startrekzky 

I just had a quick check at the default and release-v0.12 branches. In both cases the relevant code snippet (in grafana/dashboards/Gitlab.json ) was:

```sql
with _prs as(
  SELECT
    DATE_ADD(date(created_date), INTERVAL -DAY(date(created_date))+1 DAY) as time,
    avg(TIMESTAMPDIFF(Minute,created_date,merged_date)/1440) as time_to_merge
  FROM pull_requests
  WHERE
    $__timeFilter(created_date)
    and base_repo_id in ($repo_id)
    and created_date >= DATE_ADD(DATE_ADD($__timeFrom(), INTERVAL -DAY($__timeFrom())+1 DAY), INTERVAL +1 MONTH)
  group by 1
)
```

which is till grouping by the month of `create_date`. Did I look at the wrong file?
